strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/io/ArtifactOutputStreamTest.java

```diff
@@ -42,13 +42,22 @@
 {
 
     public static final String REPOSITORY_RELEASES = "aos-releases";
+    
+    private static boolean initialized;
```

Comment:
@sbespalov,

Some of the artifact/repository-related tests have to work this way. They are mostly not dependent on each other.

The tests that need repositories work like this:

* Each test class has:
  * A `public static Set<Repository> getRepositoriesToClean()` which defines the set of repositories it needs to create, or clean in order to be able to execute.
  * A `@BeforeClass public static void cleanUp() {...}` which takes care of removing the repositories. The purpose of this method is to make sure you can re-run the tests in your IDE.
  * It's own repository (or set of repositories) created during the initialization of the test. Once.
* The respective initialization method is annotated with `@PostConstruct`, it takes care of creating the test repository and populating it with the respective artifacts. Each test artifact has a comment showing which test method it belongs to, so that artifacts are not shared between the different test methods. (I know, it's not optimal, but re-initializing before every test method would make the build take three times longer. I have tried it. It's not cool at all and these are not integration tests -- they are unit tests that have to pass before committing+pushing the code).

I know. This is not optimal. However, we don't want to spend 30 minutes in one build. If you have any smarter, or simpler ideas, please share. I don't want the build to take much longer than 3-5 minutes. I've already extracted some of the tests that I know should be integration tests in #263 in order to shorten the build times.